build.sh

```diff
@@ -147,28 +164,66 @@ function get_and_build()
 	rm -rf "$tmp_path"
 }
 
-if [[ "$1" = "--list" ]]; then
-	echo "Fetching list from servers"
-	list_versions
-	exit
-elif [[ $# -lt 3 ]]; then
-	usage
-fi
+# Args handling
+opts="$(getopt -o 'h?' --long 'list,device:,os-version:,src:,dest-dir:' -- "$@" || { usage >&2 && exit 1; })"
+eval set -- "$opts"
+
+device=
+versions=
+module_dir=
+output_dir="output"
+
+while true; do
+    flag=$1
+    shift
+    case "$flag" in
+        --device) device="$1" && shift ;;
+		--os-version) versions="$1" && shift ;;
+		--dest-dir) output_dir="$1" && shift ;;
+        --src) module_dir="$1" && shift ;;
+        --) break ;;
+        *)
+            {
+                echo "error: unknown flag: $flag"
+                usage
+            } >&2
+            exit 1
+            ;;
+    esac
+done
 
-device="$1"
-version="$2"
-module_dir="$3"
+# which command
+command="$1"
+case "$command" in
+	build)
+		shift
+		;;
+	list)
+		echo "Fetching list from servers" && list_versions && exit
+		;;
+	*)
+		{
+			echo "error: unknown command: $1"
+			usage
+		} >&2
+		exit 1
+		;;
+esac
 
 [[ -d "$module_dir" ]] || fatal "ERROR: Cannot find module directory $module_dir"
+[[ -z "$device" ]] && fatal "ERROR: No device specified"
```

Comment:
If no device type is specified we should use `%%BALENA_MACHINE_NAME%%` but default and just print a warning that no device type was specified so using the default.

Reply:
Done